library/ssl_misc.h

```diff
@@ -756,7 +756,7 @@ struct mbedtls_ssl_handshake_params {
  * access to ecdh_ctx structure is needed for MBEDTLS_ECDSA_C which does not
  * seem correct.
  */
-#if defined(MBEDTLS_ECDH_C) || defined(MBEDTLS_ECDSA_C)
+#if defined(MBEDTLS_ECDH_C) || defined(MBEDTLS_PK_CAN_ECDSA_SOME)
```

Comment:
I seem to remember @AndrzejKurek pointing out that guards were inconsistent between `ssl_misc.h` and `ssl_tls.c` (I think) but I can't find the comment again.

Anyway, I have a look and it seems weird to have `|| ECDSA` here, this is about ECDH so the guard should be just ECDH. Turns out if we look a few lines up, there's a comment literally saying this does not seem correct!

Since we are touching this, we should make things right. It's not good having guards with comments saying "this does not seem correct but turned out to be necessary to avoid failures". Can you try removing `|| defined(MBEDTLS_PK_CAN_ECDSA_SOME)` then we'll see what breaks and try to fix it the right way this time?

Reply:
> I seem to remember @AndrzejKurek pointing out that guards were inconsistent between ssl_misc.h and ssl_tls.c (I think) but I can't find the comment again.

Yes it was in the previous PR.

>Turns out if we look a few lines up, there's a comment literally saying this does not seem correct! 

TBH I was going to remove that guard as well at first, but then I saw the comment and I didn't want to dig too much into it, so I left it there. I will try to remove it see how it goes. There are 2 options in my opinion:

- find a better description for this guard, or
- the guard is not necessary at all

Reply:
From a logical perspective, the `#if` condition really does not look correct: why would we need to have an ECDH context when only ECDSA is enabled? This means either something's wrong in another part of the code and we should fix it, or I'm missing some subtlety and the comment above should explain those subtleties so that we understand what's going on next time we look at this part of the code :)

Reply:
I [removed that guard](https://github.com/Mbed-TLS/mbedtls/pull/7321/commits/2032a1d5d76e18d5b57ccabfa75d8c5be0d1100e) and run `analyze_outcomes.py` and everything went smoothly so it seems that it was not necessary after all